force-app/main/default/triggers/AccountTriggerAddressCopy.trigger

```diff
@@ -0,0 +1,19 @@
+trigger AccountTriggerAddressCopy on Account (before insert) {
+    for (Account newAccount : Trigger.new) {
+        // Check if shipping address fields are not empty
+        if (!String.isBlank(newAccount.ShippingStreet) &&
+            !String.isBlank(newAccount.ShippingCity) &&
+            !String.isBlank(newAccount.ShippingState) &&
+            !String.isBlank(newAccount.ShippingPostalCode) &&
+            !String.isBlank(newAccount.ShippingCountry)) {
+                
+            // Copy shipping address to billing address
+            newAccount.BillingStreet = newAccount.ShippingStreet;
+            newAccount.BillingCity = newAccount.ShippingCity;
+            newAccount.BillingState = newAccount.ShippingState;
+            newAccount.BillingPostalCode = newAccount.ShippingPostalCode;
+            newAccount.BillingCountry = newAccount.ShippingCountry;
+        }
+    }
+}
```

Comment:
Good another option here is to use the null coalescing operator and assign a default value of an empty string if it is null 

```
  private void setDefaultBillingAddress(){
        for(Account account : accounts){
            account.BillingStreet = account.ShippingStreet ?? '';
            account.BillingCity = account.ShippingCity ?? '';
            account.BillingState = account.ShippingState ?? '';
            account.BillingPostalCode = account.ShippingPostalCode ?? '';
            account.BillingCountry = account.ShippingCountry ?? '';

        }
    }
    ```